universal-application-tool-0.0.1/app/views/BaseHtmlView.java

```diff
@@ -17,6 +27,64 @@
  */
 public abstract class BaseHtmlView {
 
+  public Tag renderHeader(String headerText) {
+    return h1(headerText);
+  }
+
+  protected ImmutableList<DomContent> textInputWithLabel(
+      String labelValue, String inputId, Optional<String> value) {
+    Tag labelTag = label(labelValue).attr("for", inputId);
+    Tag inputTag = input().withType("text").withId(inputId).withName(inputId);
+    if (value.isPresent()) {
+      inputTag.withValue(value.get());
+    }
+
+    return ImmutableList.of(labelTag, br(), inputTag, br(), br());
+  }
+
+  public ImmutableList<DomContent> textInputWithLabel(
+      String labelValue, String inputId, String value) {
+    Optional<String> optionalValue =
+        (value == null || value.isEmpty()) ? Optional.empty() : Optional.of(value);
```

Comment:
question: I think you can shorten this to `Optional.ofNullable(value)`, but I'm not sure value can/should be null, can it? (same for below)

Reply:
Replaced with `Optional<String> optionalValue = Optional.ofNullable(value).filter(s -> !s.trim().isEmpty());`